src/odemis/driver/orsay.py

```diff
@@ -434,6 +438,113 @@ def terminate(self):
 
             super(OrsayComponent, self).terminate()
 
+    # Helper functions to handle presets (on top of the preset manager)
+    # They can be used by other class of this module, but otherwise are considered private
+    def _load_preset(self, preset_name: str, preset_mask: str, timeout: float=300):
+        """
+        Load a preset, for all the device/parameters defined in the preset mask,
+        and wait until it's done.
+        :param preset_name: name of the preset
+        :param preset_mask: name of the preset mask
+        :param timeout: The maximum time to wait (s) for the preset to be loaded
+        :raises:
+            FileNotFound: if the preset or preset mask don't exist
+            RuntimeError: if other error happen during loading the preset (on the Orsay side)
+            TimeoutError: if the preset doesn't appear loaded within the specified timeout
+        """
+        tstart = time.time()
+        try:
+            # Will raise an exception if the preset or mask don't exist (since 2023-01-30)
+            self.preset_manager.LoadPreset(preset_name, preset_mask)
+        except socket.timeout:
+            # Loading can takes so long that the connection temporarily times out.
+            # That is fine, it reconnects automatically.
+            logging.debug("LoadPreset timed out, but that's expected")
+            pass
+
+        # We read the preset and preset masks for two reasons:
+        # * confirm they exists (because if they don't, LoadPreset failed silently)
+        # * wait until all the affected parameters are updated
+        try:
+            preset_content = self.preset_manager.GetPreset(preset_name)
+        except Exception:
+            logging.exception("Failed to retrieve preset %s", preset_name)
+            raise
+
+        try:
+            preset_mask_content = self.preset_manager.GetPresetMask(preset_mask)
+        except Exception:
+            logging.exception("Failed to retrieve preset mask %s", preset_mask)
+            raise
+
+        while not self._is_at_target_preset(preset_content, preset_mask_content):
+            time.sleep(0.5)
+            if time.time() > tstart + timeout:
+                raise TimeoutError(f"Setting preset {preset_name} timed out after {timeout} s")
+
+        logging.debug("Preset %s loaded", preset_name)
+
+    def _is_at_target_preset(self, preset, preset_mask) -> bool:
```

Comment:
Unused argument `preset`

Reply:
Yes, I'm aware of it, but preset+preset_mask are typically always handled together. I consider the fact that we don't need the preset just an implementation detail. There could be slightly different implementation that would need the preset, so I prefer to keep the interface as it is.